server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -457,4 +459,28 @@ public long getErrorsStat() {
   public long getCandidatesStat() {
     return candidates;
   }
+
+  @Override
+  public boolean isRootTable() {
+    return level == DataLevel.ROOT;
+  }
+
+  @Override
+  public boolean isMetadataTable() {
+    return level == DataLevel.METADATA;
+  }
+
+  @Override
+  public Set<TableId> getCandidateTableIDs() {
+    if (isRootTable()) {
+      return Collections.singleton(MetadataTable.ID);
+    } else if (isMetadataTable()) {
+      Set<TableId> tableIds = new HashSet<>(getTableIDs());
```

Comment:
This is OK but just calling `getTableIDs()` is probably not enough on a highly active cluster. This just calls the API and gets whatever is cached at the time. 

@ctubbsii comment from the original PR: "It can't detect tables that existed at the start but skipped, if the ZooCache wasn't up-to-date, and it can't detect skipping over any tables that were created or deleted during the scan. This is especially problematic if a new table was created as the result of a clone operation, which will duplicate all the file references for the new table."

Reply:
do you think this is a better implementation of `getTableIDs()`: 
```
  @Override
  public Set<TableId> getTableIDs() {
    final String tablesPath = context.getZooKeeperRoot() + Constants.ZTABLES;
    final ZooReader zr = context.getZooReader();
    final Set<TableId> tids = new HashSet<>();
    while (true) {
      try {
        zr.sync(tablesPath);
        zr.getChildren(tablesPath).forEach(t -> tids.add(TableId.of(t)));
        return tids;
      } catch (KeeperException | InterruptedException e) {
        log.error("Error getting tables from ZooKeeper, retrying", e);
        UtilWaitThread.sleepUninterruptibly(1, TimeUnit.SECONDS);
        continue;
      }
    }
  }
```